core/src/main/scala/better/files/ThreadBackedFileMonitor.scala

```diff
@@ -60,15 +57,12 @@ abstract class ThreadBackedFileMonitor(val root: File, maxDepth: Int) extends Fi
     }
   }
 
-  override def start() = {
+  override def start()(implicit executionContext: ExecutionContext) = {
     watch(root, maxDepth)
-    thread.start()
+    executionContext.execute(() => Iterator.continually(service.take()).foreach(process))
```

Comment:
Yup, but when you call start on the monitor, it'll be called just once no? And setup a thread to `take` events from the `WatchService`, there's only one `WatchServer` isn't there?